misc/helper-scripts/rosmsg_to_asn1.py

```diff
@@ -0,0 +1,417 @@
+    :return slot_type: The base data type of the slot (if it is an array).
+    :return field_type: The field_type of the slot.
+    :return length: The length of the array.
+    """
+    bracket = "["
+    length = 1
+    field_type = "simple"
+    idx = slot_type.find(bracket)
+    if idx == -1:
+        return slot_type, field_type, length
+    else:
+        if slot_type[idx + 1] == "]":
+            field_type = "variable-length"
+        else:
+            field_type = "fixed-length"
+            idx_end = slot_type.find("]")
+            str_length = slot_type[idx+1:idx_end]
+            length = int(str_length)
+        slot_type = slot_type[0:idx]
+    return slot_type, field_type, length
+
+
+def find_ASN1_type(slot_type):
+    """
+    Given the ROS type of a field finds and returns the equivalent ASN.1 type.
+    If the type is not primitive a flag is set so that a dependency is
+    generated.
+    :param slot_type: The ROS field type.
+    :return asn1_type: The ASN.1 equivalent type of the ROS type.
+    :return lib_asn: The library the ASN.1 type belongs to.
+    :return primitive: Flag to denote if the field type is primitive.
+    """
+    primitive = False
+    if slot_type in ros_primitive_type:
+        asn1_type = primitive_types_map[slot_type]
+        lib_asn = libraries[asn1_type]
+        primitive = True
+    else:
+        composed_type = slot_type.split("/")
+        if len(composed_type) == 1:
+            logging.warn("The length of the composed_type is 1!")
+        asn1_type = composed_type[1]
+        # primitive_types_map[slot_type] = asn1_type
+        lib_asn = composed_type[1]+'-Types'
+        # libraries[asn1_type] = lib_asn
+        primitive = False
+    return asn1_type, lib_asn, primitive
+
+
+def generate_libraries_string(import_libraries):
+    """
+    Generate a string from a set of libraries to be imported to the ASN.1 type.
+    :param import_libraries: The libraries to be imported.
+    :return s: The libraries names in string format.
+    """
+    s = "IMPORTS "
+    comma = ""
+    for key in import_libraries:
+        num_types = len(import_libraries[key])
+        for t in import_libraries[key]:
+            s += comma + t
+            comma = ", "
+        s += " FROM " + key + " "
+        comma = ""
+    s += ";"
+    s += "\n"
+    return s
+
+
+def process_msg(rospack, msg):
+    """
+    Process a message creating a string containing an ASN.1 representation.
+    :param rospack: A rospkg instance.
+    :param msg: The ROS message to be converted to ASN.1 message.
+    :return msg_string: The string containing the ASN.1 representation.
+    :return msg_deps: Any message dependencies that need to be generated.
+    """
+    msg_name = None
+    pkg_name = None
+    msg_string = None
+    msg_deps = None
+    # Get the package contining the message
+    msg_name, pkg_name = get_message_package(rospack, msg)
+    if msg_name is None or pkg_name is None:
+        return msg_string, msg_deps
```

Comment:
Prefer `return None, None` here and below. Consider adding a blank line after this line.

Reply:
Fixed